internal/scheduler/metrics/metrics.go

```diff
@@ -273,8 +268,14 @@ func (m *Metrics) UpdateCancelled(job *jobdb.Job) error {
 	return nil
 }
 
-func (m *Metrics) UpdateFailed(ctx *armadacontext.Context, job *jobdb.Job, jobRunErrorsByRunId map[uuid.UUID]*armadaevents.Error) error {
+func (m *Metrics) UpdateFailed(ctx *armadacontext.Context, job *jobdb.Job, jctx *schedulercontext.JobSchedulingContext, jobRunErrorsByRunId map[uuid.UUID]*armadaevents.Error) error {
```

Comment:
Why do we need to pass in both a job and jctx, which has an embedded job?

Reply:
because UpdatedPreempted needed jctx. I removed jctx now it as it no longer needed for both UpdateFailed and UpdatePreempted